src/extension/server/runmeServer.ts

```diff
@@ -96,16 +129,46 @@ class RunmeServer implements Disposable {
         })
     }
 
-    async launch(): Promise<object | number | RunmeServerError> {
+    async acceptsConnection(): Promise<void> {
```

Comment:
I think upon further investigation this may be an issue with the go side of things. I think that the "`started listening`" message gets sent too early. I'll look into this more, but for now this solution is fine.

Reply:
It seemed like the singleton approach was prone to race conditions anyways so I wanted to make this more resilient. However, I do agree, if there's a way to have the message convey actual service readiness, it'd would cut down on retries and inherent inefficiencies. If it's not an easy fix, track a new ticket?